editor/editor_dock_manager.cpp

```diff
@@ -52,6 +53,118 @@ enum class TabStyle {
 
 EditorDockManager *EditorDockManager::singleton = nullptr;
 
+bool EditorDockDragHint::can_drop_data(const Point2 &p_point, const Variant &p_data) const {
+	return can_drop_dock;
+}
+
+void EditorDockDragHint::drop_data(const Point2 &p_point, const Variant &p_data) {
+	if (can_drop_dock) {
+		// Drop dock into last spot if not over tabbar.
+		if (drop_tabbar->get_rect().has_point(p_point)) {
+			drop_tabbar->_handle_drop_data("tab_container_tab", p_point, p_data, callable_mp(this, &EditorDockDragHint::_drag_move_tab), callable_mp(this, &EditorDockDragHint::_drag_move_tab_from));
+		} else {
+			dock_manager->_move_dock(dock_manager->_get_dock_tab_dragged(), dock_manager->dock_slot[occupied_slot], drop_tabbar->get_tab_count());
+		}
+	}
+}
+
+void EditorDockDragHint::_drag_move_tab(int p_from_index, int p_to_index) {
+	dock_manager->_move_dock_tab_index(dock_manager->_get_dock_tab_dragged(), p_to_index, true);
+}
+
+void EditorDockDragHint::_drag_move_tab_from(TabBar *p_from_tabbar, int p_from_index, int p_to_index) {
+	dock_manager->_move_dock(dock_manager->_get_dock_tab_dragged(), dock_manager->dock_slot[occupied_slot], p_to_index);
+}
+
+void EditorDockDragHint::gui_input(const Ref<InputEvent> &p_event) {
+	ERR_FAIL_COND(p_event.is_null());
+
+	Ref<InputEventMouseMotion> mm = p_event;
+	if (mm.is_valid()) {
+		Point2 pos = mm->get_position();
+
+		// Redraw when inside the tabbar and just exited.
+		if (mouse_inside_tabbar) {
+			queue_redraw();
+		}
+		mouse_inside_tabbar = drop_tabbar->get_rect().has_point(pos);
+	}
+}
+
+void EditorDockDragHint::set_slot(EditorDockManager::DockSlot p_slot) {
+	occupied_slot = p_slot;
+	drop_tabbar = dock_manager->dock_slot[occupied_slot]->get_tab_bar();
+}
+
+void EditorDockDragHint::_notification(int p_what) {
+	switch (p_what) {
+		case EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED: {
+			if (EditorSettings::get_singleton()->check_changed_settings_in_group("interface/theme")) {
+				dock_drop_highlight->set_corner_radius_all(EDSCALE * EDITOR_GET("interface/theme/corner_radius").operator int());
+				if (mouse_inside) {
+					queue_redraw();
+				}
+			}
+		} break;
+
+		case NOTIFICATION_THEME_CHANGED: {
+			valid_drop_color = get_theme_color(SNAME("accent_color"), EditorStringName(Editor));
+			invalid_drop_color = get_theme_color(SNAME("error_color"), EditorStringName(Editor));
+		} break;
+
+		case NOTIFICATION_MOUSE_ENTER:
+		case NOTIFICATION_MOUSE_EXIT: {
+			mouse_inside = p_what == NOTIFICATION_MOUSE_ENTER;
+			queue_redraw();
+		} break;
+
+		case NOTIFICATION_DRAG_BEGIN: {
+			Control *dragged_dock = dock_manager->_get_dock_tab_dragged();
+			if (!dragged_dock) {
+				return;
+			}
+
+			// TODO: Update logic when GH-106503 is merged to use flags.
+			can_drop_dock = is_layout_horizontal ? bool(dragged_dock->call("_can_dock_horizontal")) : true;
```

Comment:
is_layout_horizontal is always false, so this check doesn't do anything and invalid color is never used, even after #106503. 
We can just set can_drop_dock to true, and remove the invalid color for now. When a PR adds horizontal slots it can be re-added then, if it is wanted.